src/test/java/com/google/step/servlets/PromoCodeServletTest.java

```diff
@@ -89,6 +92,25 @@ public void correctChannelIdRequest() throws IOException {
                         + DateFormat.getDateTimeInstance().format(MOCK_DATE) + "\"}]\n"));
     }
 
+    @Test
+    public void correctUserIdRequest() throws IOException {
+        when(request.getParameter("formInput")).thenReturn(COMPLETE_USERNAME);
+
+        StringWriter sw = new StringWriter();
+        PrintWriter pw = new PrintWriter(sw);
+
+        when(response.getWriter()).thenReturn(pw);
+        when(infoScraper.scrapeUserUploadPlaylist(USERNAME)).thenReturn(Optional.of(UPLOAD_ID));
+        when(infoScraper.scrapePromoCodesFromPlaylist(UPLOAD_ID)).thenReturn(
+                Optional.of(Arrays.asList(PromoCode.create(CHANNEL_NAME, VIDEO_ID, VIDEO_TITLE, MOCK_DATE))));
```

Comment:
Change MOCK_DATE to just DATE.  We aren't actually mocking a Date object. Just testing it.